### ISSUE TITLE
AWS credentials validation 

### ISSUE THREAD

**Issue by samuel-br** [env0/terraform-provider-env0#515]
When we creating AWS credentials that include `External Id` we need to validate this property  as [AWS DOCS](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html) describe.

the regex we apply according to the DOCS above is   `/^[A-Za-z0-9+=\-.,@:/]{2,1224}$/`

So the expected behavior  - for example : 
```
resource "env0_aws_credentials" "my_role_by_arn" {
  name        = "Test Role arn ${random_string.random.result}"
  arn         = "Role ARN"
  external_id = "External id"
}
```
will fail because the white space `external_id = "External id"`.

This example will work
 ```
resource "env0_aws_credentials" "my_role_by_arn" {
  name        = "Test Role arn ${random_string.random.result}"
  arn         = "Role ARN"
  external_id = "External-id"
}
```
to summarise - make sure `external id` is correct as the requires of AWS

**Comment by TomerHeber** [comment]
Deferred - feature removed in the backend.

**Comment by TomerHeber** [comment]
@yaronya - I had a PR for this, but it was closed. Please let me know how to proceed. Thanks!
